various layout works at Subset tab

moontrip opened the pull request:
This PR addresses following issues:

- #381 
![change-question-tooltip-text](https://user-images.githubusercontent.com/12802305/135916738-d9c7dedf-bfe2-4970-9171-fe890e97ae2b.png)

- #289 
Sheena approved my mockup - discussed at slack

- #290 
a) change the text style
b) add show more/less link for multifilter case
Discussed with Sheena and she chose one of my mockup, which is included in this PR. See more details at the corresponding issue, #290 with screenshots

jtlong3rd commented on src/lib/workspace/Variable.tsx:
```diff
@@ -31,32 +32,88 @@ export function VariableDetails(props: Props) {
+          &nbsp;&nbsp;
+          {MultiFilterVariable.is(variable) ? (
+            <>
+              <a
+                className="variable-show-more-link"
+                style={{ cursor: 'pointer' }}
+                onClick={() => {
+                  setShowMore(!showMore);
+                }}
+              >
+                {showMoreLink}
+              </a>
```
Instead of rendering an `a`nchor with an inline `cursor: pointer` style, you can use a `button` with a `link` class:

```suggestion
              <button
                className="variable-show-more-link link"
                onClick={() => {
                  setShowMore(!showMore);
                }}
              >
                {showMoreLink}
              </button>
```

("link" is a utility class offered by [@veupathdb/web-common](https://github.com/VEuPathDB/EbrcWebsiteCommon/blob/6a4cd50198a0aaa17c78961716e9975ea6afbef9/Client/src/styles/AllSites.scss#L93) which can be used to style a button as though it were a link.)

moontrip replied:
@jtlong3rd thank you for your review and suggestions 👍 Actually I was also thinking of the use of button but just did it with <a> 😄 Will change it to <button> as suggested

jtlong3rd commented on src/lib/workspace/Variable.tsx:
```diff
@@ -31,32 +32,88 @@ export function VariableDetails(props: Props) {
+            into ClinEpiDB
           </HelpIcon>
+          &nbsp;&nbsp;
+          {MultiFilterVariable.is(variable) ? (
```
I'd add some logic to only display the "show more" if there are more than 3 original variable names. (I'm thinking about things like India ICEMR Cohort's "resident status.")

moontrip replied:
yes I also thought of that edge case but didn't make it. I totally agree that it needs to be checked just in case!

jtlong3rd reviewed (commented):
I have just two small inline suggestions, and then this should be good to merge!

moontrip commented:
@jtlong3rd I have addressed your feedbacks. For the second one, I made a variable to check the array length of the multifilter variable and use it as additional condition to display `Show more` link. An example screenshot for the case of `resident status` you suggested, which should not have the expansion link. 

![multifilter-variable-two-variables](https://user-images.githubusercontent.com/12802305/135933053-07ffff32-7b0d-4d2d-9f7a-b1e1cbb15d33.png)

jtlong3rd commented on src/lib/workspace/Variable.tsx:
```diff
@@ -31,32 +32,95 @@ export function VariableDetails(props: Props) {
+    ? findMultifilterVariableLeaves(
+        variable,
+        groupBy(entity.variables, (variable) => variable.parentId)
+      ).map((variable, i) => {
```
The linter points out: `Unnecessary escape character: \[`. Could you fix? (You can probably just replace `\]` with `]`.)

The linter also points out `Array.prototype.map() expects a value to be returned at the end of arrow function`. To fix that, I would return either `null` or  `<></>` when `i >= 3`.

moontrip replied:
@jtlong3rd you are always welcome to bother me 😄 
Anyway, actually I knew that the first warning but forgot to address it. Yes it works by simply removing the backslash. As for the second, I added `return null;` for those cases

dmfalke replied:
Wouldn't it make more sense to only `map` the first three items in the array? It's fewer iterations and clearer, imo.

```typescript
findMultifilterVariableLeaves(
  variable,
  groupBy(entity.variables, (variable) => variable.parentId)
)
.slice(3)
.map((variable, i) => {
```

jtlong3rd replied:
I agree with @dmfalke here - you could `.slice(0, 3)` to map the first three elements.

dmfalke replied:
Thank you for the clarification @jtlong3rd~

jtlong3rd commented on src/lib/workspace/Variable.tsx:
```diff
@@ -31,32 +32,95 @@ export function VariableDetails(props: Props) {
+    ? findMultifilterVariableLeaves(
+        variable,
+        groupBy(entity.variables, (variable) => variable.parentId)
+      ).map((variable, i) => {
```
The linter also throws similar warnings here.

jtlong3rd replied:
Similarly, you could `.slice(3)` to map all but the first three elements.

jtlong3rd reviewed (commented):
Two more inline suggestions, and then I'll leave you alone - honest! 😁

dmfalke reviewed (approved):
I left a follow-up comment to @jtlong3rd's inline comment. Otherwise, looks great.
